irc_send.c

```diff
@@ -213,13 +213,7 @@ void irc_send_names(irc_channel_t *ic)
 			*namelist = 0;
 		}
 
-		if (icu->flags & IRC_CHANNEL_USER_OP) {
-			strcat(namelist, "@");
-		} else if (icu->flags & IRC_CHANNEL_USER_HALFOP) {
-			strcat(namelist, "%");
-		} else if (icu->flags & IRC_CHANNEL_USER_VOICE) {
-			strcat(namelist, "+");
-		}
+		namelist[strlen(namelist)] = irc_channel_user_get_prefix(icu);
```

Comment:
Hmm, I know `array = {0}` is be guaranteed to initialize a whole array to 0 [citation needed], and technically "" is the same thing but... yeah, not sure.

Same thing applies to the old strcat() code though.

Could certainly use GStrings in this code to make it cleaner, if we don't mind allocating some more stuff in the heap instead of the stack. Probably not worth changing though.